python/docs/source/development/contributing.rst

```diff
@@ -189,9 +186,6 @@ Annotations should, when possible:
 
 * Be compatible with the current stable MyPy release.
 
-
-Complex supporting type definitions, should be placed in dedicated ``_typing.pyi`` stubs. See for example `pyspark.sql._typing.pyi <https://github.com/apache/spark/blob/master/python/pyspark/sql/_typing.pyi>`_.
```

Comment:
One more question ‒ do we intentionally remove this @itholic?

Reply:
Checking in on this last question @itholic

Reply:
Yeah, this is intentionally removed since I think we better not to mention about the stub files (pyi) in the guide.

Do you think it's good to keep this part as it is? (I have no strong opinion in either way, though)

Reply:
`_typing` modules are here to stay as they're fundamental part of our annotations so I don't think we should ignore them in the typing guide.

As of mentioning stubs explicitly I don't have strong opinion. We won't be able to switch to plain Python modules before the successor of 3.4 the earliest (we need to drop Python 3.7 to support `Protocol` and `Literal` if I recall correctly) assuming we won't use any new typing features and cyclic dependencies won't cause too much problems for migration. However, it is not necessary to mention that we use specifically use stubs.

Reply:
Got it. Let's keep this as it is for now.